### PR TITLE
Simplify checkIsHoverable

### PR DESCRIPTION
**Pull request by UselessToucan**

### CODE REVIEW

File: osu.Framework/Input/InputManager.cs [hunk 1]
```diff
@@ -105,10 +105,8 @@ protected override void Update()
             List<InputState> pendingStates = new List<InputState>();
             foreach (var h in inputHandlers)
             {
-                if (!h.IsActive) continue;
-
-                foreach (var s in h.GetPendingStates())
-                    pendingStates.Add(s);
+                if (h.IsActive)
+                    pendingStates.AddRange(h.GetPendingStates());
```
Comment (peppy) [inline osu.Framework/Input/InputManager.cs]: does addrange work okay for an empty list?

File: osu.Framework/Input/InputManager.cs [hunk 2]
```diff
@@ -208,16 +206,7 @@ private void buildMouseInputQueue(InputState state, Drawable current)
...
-        {
-            if (!d.HandleInput || !d.IsVisible || d.IsMaskedAway)
-                return false;
-
-            if (!d.Contains(state.Mouse.Position))
-                return false;
-
-            return true;
-        }
+        private bool checkIsHoverable(Drawable d, InputState state) => d.HandleInput && d.IsVisible && !d.IsMaskedAway && d.Contains(state.Mouse.Position);
```
Comment (peppy) [inline osu.Framework/Input/InputManager.cs]: not sure i prefer this

File: osu.Framework/Input/InputManager.cs (hunk 1 above)
Comment (UselessToucan) [inline osu.Framework/Input/InputManager.cs]: It works ok for an empty list. (but not for `null`)

File: osu.Framework/Input/InputManager.cs (hunk 2 above)
Comment (Tom94) [inline osu.Framework/Input/InputManager.cs]: I definitely do. For me the negation with || is way harder to parse mentally than a composition via &&. Its just like ticking off a checklist, making sure all conditions are met.

**Pull request merged**